### PR TITLE
[post] First Draft - Schematic Review Checklist

### PR DESCRIPTION
**Pull request by schultetwin**
Add the first draft of a firmware engineer's guide to schematic review.

### CODE REVIEW

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:76]: Would you say there are other recommended options for testing these pieces, or is this the best approach? Wondering if you should remove the second sentence here and give a stronger opinion on needing test points

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:77]: I'm wondering if you might consider another section dedicated to memory with the bottom line being that you should have plenty of space for extra code you'll add in the future, crash data (so coredumps), and logs. Perhaps you have a story about not having enough flash when you wanted it, and you wish you fought harder for a bigger flash part

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:106]: This sounds great, and a forcing function for both the fw engineer and the EE to make sure you agree on the GPIO requirements. Schematics are very dense so breaking out this information makes complete sense!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:109]: Out of curiosity, was his concern at all related to tampering, so he wanted to make it "hard" for them? (of course they would probably just solder it themselves if they got as far as breaking it open). Or just make it hard for you 😄

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:57]: I'd add the detail you had in your story here so it is as complete as possible
```suggestion
- [ ] Provide a GPIO map in a spreadsheet with main function after reset, intended function, and any additional notes 
```

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:51]: To clarify, is this asking "how does the firmware shut off power to the device, particularly if it is battery-powered?" or "how does the MCU get power from the power source, particularly if it is a battery?" I'm guessing the first but want to be sure because "power off" could have a few interpretations

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:117]: If you are interested, you could ask readers to comment what their schematic review horror stories are and what they now always check for

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:70]: Could also add checking your typical Tx -> Rx in addition to flow control lines 😄

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:132]: We can remove this section if you don't have any references you'd like to link

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [inline _drafts/schematic_review_checklist_for_firmware_engineer.md:46]: What do you think of doing stories first, and then give the checklist? I think that would give context to the items in your list, and act as a nice summary of sorts

**Review by gminn — CHANGES_REQUESTED** [review]
Hi @schultetwin! This is an awesome article. I have some questions and suggestions to spruce it up before we merge. Let me know if I can clarify anything!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:94]: I cleaned up this section to focus more on the fact that we couldn't turn it off, and less on the low-power mode.

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:76]: Good point! I just removed the second sentence to make this a stronger statement.

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:77]: Added this section!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:109]: I don't think so... I don't want to speak for this other engineer, but my thought is, (as you stated) if you're reverse engineering a product, attaching leads to JTAG is likely a small step.

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:89]: Added a concluding sentence here!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:51]: Switched wording to "shutdown".

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:117]: Done! Thank you!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:70]: Yeah, good point :). I've seen that messed up a bunch. Added!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (schultetwin) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md:46]: Done!

File: _drafts/schematic_review_checklist_for_firmware_engineer.md
Comment (gminn) [reply _drafts/schematic_review_checklist_for_firmware_engineer.md]: Makes sense, thanks for the clarification!

**Review by gminn — APPROVED** [review]
Thank you @schultetwin for addressing my comments! Really excited for this one. I'll merge this draft and then when we are looking to publish, I'll open a publish PR for you to review.

**Pull request merged**
